contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -14,92 +14,339 @@
-  static std::string empty;
-  return empty;
+Status AuthzChecker::ParseTestResponse(
+    std::shared_ptr<context::RequestContext> context,
+    const std::string &json_str) {
+  grpc_json *json = grpc_json_parse_string_with_len(
+      const_cast<char *>(json_str.data()), json_str.length());
+
+
+  if (!json) {
+    return Status(Code::INVALID_ARGUMENT,
+                  "Invalid JSON response from Firebase Service");
+  }
+
+  Status status = Status::OK;
+  Status invalid = Status(Code::INTERNAL, kInvalidResponse);
+
+  const grpc_json *testResults = GetProperty(json, "testResults");
+  if (testResults == nullptr) {
+    env_->LogError("TestResults are null");
+    status = invalid;
+  } else {
+    const char *result = GetStringValue(testResults->child, "state");
+    if (result == nullptr) {
+      env_->LogInfo("Result state is empty");
+      status = invalid;
+    } else if (std::string(result) != "SUCCESS") {
+      status = Status(Code::PERMISSION_DENIED,
+              std::string("Unauthorized ")
+              + context->request()->GetRequestHTTPMethod()
+              + " access to resource " + context->request()->GetRequestPath(),
+              Status::AUTH);
+    }
+  }
+
+  grpc_json_destroy(json);
+  return status;
+}
+
+std::string AuthzChecker::GetOperation(std::string httpMethod) {
+  if (httpMethod == "POST") {
+    return std::string("create");
+  }
+
+  if (httpMethod == "GET" || httpMethod == "HEAD" || httpMethod == "OPTIONS") {
+    return std::string("get");
+  }
+
+  if (httpMethod == "DELETE") {
+    return std::string("delete");
+  }
+
+  return std::string("update");
+}
+
+std::string AuthzChecker::BuildTestRequestBody(
+    std::shared_ptr<context::RequestContext> context) {
+
+  std::ostringstream ss;
+
+  int openCount = 0;
+  ss << "{"
+     <<   Stringify("test_cases") + ": "
+     <<     "[ {";
+  ++openCount;
+  AddToBody("service_name", context->service_context()->service_name(), false, ss);
+  AddToBody("resource_path", context->request()->GetRequestPath(),
+            false, ss);
+  AddToBody("operation", GetOperation(context->request()->GetRequestHTTPMethod()),
+            false, ss);
+  AddToBody("expectation", "ALLOW", false, ss);
+  AddToBody("variables", ss);
+  ++openCount;
+  AddToBody("request", ss);
+  ++openCount;
+  AddToBody("auth", ss);
+  ++openCount;
+
+  ss << Stringify("token") + ": " << context->auth_claims();
+
+  while(openCount-- > 0) {
+    ss << "} ";
+  }
+
+  ss << "] }";
+  return ss.str();
+}
+
+void AuthzChecker::AddToBody(const std::string &key,
+                       std::ostringstream &ss) {
+  ss << Stringify(key.c_str()) + ": " + "{";
+}
+void AuthzChecker::AddToBody(const std::string &key, const std::string &value,
+                             bool end, std::ostringstream &ss) {
+  ss << Stringify(key.c_str()) + ": " + Stringify(value.c_str());
+  if (!end) {
+    ss << ", ";
+  }
+}
+
+const std::string AuthzChecker::GetReleaseName(
```

Comment:
Same reason as above. There is no meaning to this method outside the context of the class.